Source/SwiftLintFramework/Rules/ExplictSelfRule.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  ExplictSelfRule.swift
+//  SwiftLint
+//
+//  Created by Ian Keen on 2016-10-06.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct ExplicitSelfRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.Error)
+
+    public init() { }
+
+    public static let description = RuleDescription(
+        identifier: "explicitSelf",
+        name: "Explict Self",
+        description: "Require explicit self for instance members",
+        nonTriggeringExamples: [
+            "class Good1 { let value: Int = 42; var property: Int { return self.value } }",
```

Comment:
And maybe same for accessing variables in a scope other than a type (like global values or local variables in a global function scope.
Also `static` functions! (Sorry, just thinking of more as I go hehe)